lib/libutee/tee_api_operations.c

```diff
@@ -763,39 +763,17 @@ static TEE_Result tee_buffer_update(
 
 	if (slen >= (buffer_size + buffer_left)) {
 		/* Buffer is empty, feed as much as possible from src */
-		if (TEE_ALIGNMENT_IS_OK(src, uint32_t)) {
```

Comment:
`tee_buffer_update()` is calling `utee_cipher_update()`. It is not specified that this takes aligned buffers as inputs

This will then call `crypto_ops.cipher.update`, which also takes `uint8_t *` pointer for data and output.

I think we used to need it because of the old library crypto library, before the libtomcrypt use. If someone is using a library requiring specific alignment, I think he will have to do the job internally.